dataprep/eda/missing/compute.py

```diff
@@ -115,10 +116,86 @@ def missing_spectrum(df: dd.DataFrame, bins: int, ncols: int) -> Intermediate:
         }
     )
     return Intermediate(
-        data=df, missing_percent=missing_percent, visual_type="missing_spectrum"
+        data=df, missing_percent=missing_percent, visual_type="missing_spectrum",
     )
 
 
+def missing_spectrum_tabs(df: dd.DataFrame, bins: int, ncols: int) -> Intermediate:
+    """
+    Calculate a heatmap visualization of nullity correlation in the given DataFrame
+    """
+    # pylint: disable=too-many-locals
+    ### Song: if you output the df here,
+    ### You would find that the size of df is not different from the df you output in the line 159
+    ### print(df.shape) here
+    ### You could find that the variable df is the original dataframe.
+    df1 = df.compute()
+    df2 = df.compute()
+    length = len(df2)
+    num_bins = min(bins, len(df) - 1)
+
+    df = df.iloc[:, :ncols]
+    cols = df.columns[:ncols]
+    ncols = len(cols)
+    nrows = len(df)
+    chunk_size = len(df) // num_bins
+    data = df.isnull().to_dask_array()
+    data.compute_chunk_sizes()
+    data = data.rechunk((chunk_size, None))
+
+    (notnull_counts,) = dd.compute(data.sum(axis=0) / data.shape[0])
+    missing_percent = {col: notnull_counts[idx] for idx, col in enumerate(cols)}
+
+    missing_percs = data.map_blocks(missing_perc_blockwise, dtype=float).compute()
+    locs0 = np.arange(len(missing_percs)) * chunk_size
```

Comment:
locs0 does not make sense here. Could you give it a meaningful name? The same problem also exists in the following code.